multi-agent/lib/mas/resources/builtin_aware_service.py

```
@@ -0,0 +1,532 @@
+        rid: str,
+        caller: CallerScope = CallerScope(),
+    ) -> ElementCard:
+        cards = self.get_cards([rid], caller=caller)
+        if rid not in cards:
+            raise KeyError(f"Resource not found: {rid}")
+        return cards[rid]
+
+    # ── Overridden internal method ───────────────────────────────────────
+
+    def _build_configs_from_rids(
+        self, rids: List[str], caller: CallerScope = CallerScope(),
```

Ruff (0.16.1)

[warning] Do not perform function call `CallerScope` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)